compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -277,13 +285,14 @@ where
                     // We didn't find the query result in the query cache. Check if it was
                     // poisoned due to a panic instead.
                     let lock = query.query_state(qcx).active.get_shard_by_value(&key).lock();
-                    match lock.get(&key) {
-                        // The query we waited on panicked. Continue unwinding here.
-                        Some(QueryResult::Poisoned) => FatalError.raise(),
-                        _ => panic!(
-                            "query result must in the cache or the query must be poisoned after a wait"
-                        ),
+
+                    if let Some(QueryResult::Poisoned) = lock.get(&key) {
```

Comment:
Why did you change this into an `if let`? A match is easier to read.

Reply:
https://github.com/rust-lang/rust/blob/0919a5145d412c3ff3ca12e315610417fc1e3364/compiler/rustc_query_system/src/query/plumbing.rs#L284-L294

Does not type check. But can just put the panic back.